applications/crossbar/priv/couchdb/account/rate_limits.json

```diff
@@ -0,0 +1,12 @@
+{
+    "_id": "_design/rate_limits"
+    ,"language": "javascript"
+    ,"views": {
+        "crossbar_listing": {
+            "map": "function(doc) {if (doc.pvt_type != 'rate_limits' || doc.pvt_deleted) return;var emit_records = function(Name, Type, Limits) {for (SipMethod in Limits.per_minute) {emit([Name, SipMethod], {'type': Type,'per_minute': Limits.per_minute[SipMethod]})};for (SipMethod in Limits.per_second) {emit([Name, SipMethod], {'type': Type,'per_second': Limits.per_second[SipMethod]})};};if (doc.pvt_owner_type == 'account') {emit_records('device-default-rate-limits', 'device', doc.device);emit_records(doc.pvt_queryname, 'realm', doc.account);}if (doc.pvt_owner_type == 'device') emit_records(doc.pvt_queryname, 'device', doc);};"
+        },
+        "list_by_owner": {
+            "map":"function(doc) {if (doc.pvt_type != 'rate_limits' || doc.pvt_deleted) return; emit(doc.pvt_owner_id, doc._id || doc.id);}"
```

Comment:
`doc._id` will always be present, so not sure why you have `doc.id`. Also, you will get the ID in the view results, so I don't know that you need to emit it here; you could emit `null` instead.